onnxscript/ir/_core_test.py

```diff
@@ -797,6 +797,154 @@ def test_remove_safe_removes_uses_of_removed_nodes(self):
         self.assertEqual(add_node.inputs, (None, None))
 
     # TODO(justinchuby): Test graph mutation methods
+    def test_topological_sort_empty(self):
+        graph = _core.Graph(
+            inputs=(),
+            outputs=(),
+            nodes=(),
+            opset_imports={"": 1},
+        )
+        graph.sort()
+        self.assertEqual(tuple(graph), ())
+
+    def test_topological_sort_linear(self):
+        # edges=[(1,2),(2,3)]
+        v0 = _core.Value(name="v0")
+        node1 = _core.Node("", "Node1", inputs=(v0,), num_outputs=1)
+        node2 = _core.Node("", "Node2", inputs=(node1.outputs[0],), num_outputs=1)
+        node3 = _core.Node("", "Node3", inputs=(node2.outputs[0],), num_outputs=1)
+        graph = _core.Graph(
+            (v0,),
+            node3.outputs,
+            nodes=(node3, node2, node1),
+            opset_imports={"": 1},
+        )
+        graph.sort()
+        sorted_nodes = tuple(graph)
+        expected_order = (node1, node2, node3)
+        self.assertEqual(sorted_nodes, expected_order)
+
+    def test_topological_sort_independent_subgraphs(self):
+        # edges=[(1,3),(2,4)]
+        v0 = _core.Value(name="v0")
+        v1 = _core.Value(name="v1")
+        node1 = _core.Node("", "Node1", inputs=(v0,), num_outputs=1)
+        node2 = _core.Node("", "Node2", inputs=(v1,), num_outputs=1)
+        node3 = _core.Node("", "Node3", inputs=(node1.outputs[0],), num_outputs=1)
+        node4 = _core.Node("", "Node4", inputs=(node2.outputs[0],), num_outputs=1)
+        graph = _core.Graph(
+            (v0, v1),
+            (node3.outputs[0], node4.outputs[0]),
+            nodes=(node4, node3, node2, node1),
+            opset_imports={"": 1},
+        )
+        graph.sort()
+        sorted_nodes = tuple(graph)
+        expected_order = (node2, node4, node1, node3)
+        self.assertEqual(sorted_nodes, expected_order)
+
+    def test_topological_sort_shared_child(self):
+        # edges=[(1,3),(2,3)]
```

Comment:
I would create a test to verify that 

`(node3, node2, node1)` yeilds `(node2, node1, node3)`

and 

`(node3, node1, node2)` yeilds `(node1, node2, node3)`

to verify the stability claim.

Also test `(node1, node3, node2)` and `(node2, node3, node1)` 

as well as `(node1, node2, node3)` (the sorted case)